src/client/util.rs

```diff
@@ -0,0 +1,27 @@
+use tower_service::Service;
+use futures::Future;
+use tokio_io::{AsyncRead, AsyncWrite};
+
+pub trait ConnectService<A>{
+    type Response: AsyncRead + AsyncWrite;
+    type Error;
+    type Future: Future<Item =  Self::Response, Error = Self::Error>;
+
+    fn connect(&self) -> Self::Future;
```

Comment:
You probably want to take an argument of `A` here, to allow passing in the SocketAddr.

Reply:
Right, that was my original plan but in the test and example the address is contained within the "service".

https://github.com/tower-rs/tower-h2/blob/master/examples/client.rs#L48